java/ql/lib/semmle/code/java/security/XxeQuery.qll

```diff
@@ -0,0 +1,55 @@
+/** Provides taint tracking configurations to be used in XXE queries. */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.dataflow.TaintTracking2
+import semmle.code.java.security.XmlParsers
+import DataFlow::PathGraph
+
+/**
+ * A taint-tracking configuration for unvalidated remote user input that is used in XML external entity expansion.
+ */
+class XxeConfig extends TaintTracking::Configuration {
+  XxeConfig() { this = "XxeConfig" }
+
+  override predicate isSource(DataFlow::Node src) { src instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof UnsafeXxeSink }
+}
+
+/**
+ * A taint-tracking configuration for unvalidated local user input that is used in XML external entity expansion.
+ */
+class XxeLocalConfig extends TaintTracking::Configuration {
```

Comment:
Hold, wait! You can't put both of these configurations in the same file - that'll ruin performance of the regular XXE query,

Reply:
🤦‍♂️ That was a silly mistake, sorry. I'll split that up.